contracts/FastEcMul.sol

```diff
@@ -413,30 +411,28 @@ library FastEcMul {
     }
     shift = 256 - shift;
     aM = (_aM << shift) + (shift > 128 ? _am << (shift - 128) : _am >> (128 - shift));
-    uint256 a0 = (_am << shift) & 2**128-1;
-    uint256[2] memory b;
+    uint256 a0 = (_am << shift) & TWO_TO_THE_ONE_HUNDRED_TWENTY_EIGHT_MINUS_ONE;
 
-    (b[1], b[0]) = ((_b << shift) >> 128, (_b << shift) & 2**128-1);
+    (uint256 b1, uint256 b0) = ((_b << shift) >> 128, (_b << shift) & TWO_TO_THE_ONE_HUNDRED_TWENTY_EIGHT_MINUS_ONE);
 
     uint256 rM;
-    uint256 q = aM / b[1];
-    rM = aM % b[1];
+    uint256 q = aM / b1;
+    rM = aM % b1;
 
-    uint256 rsub0 = (q & 2**128-1) * b[0];
-    uint256 rsub21 = (q >> 128) * b[0] + (rsub0 >> 128);
-    rsub0 &= 2**128-1;
+    uint256 rsub0 = (q & TWO_TO_THE_ONE_HUNDRED_TWENTY_EIGHT_MINUS_ONE) * b0;
+    uint256 rsub21 = (q >> 128) * b0 + (rsub0 >> 128);
+    rsub0 &= TWO_TO_THE_ONE_HUNDRED_TWENTY_EIGHT_MINUS_ONE;
 
     while (rsub21 > rM || rsub21 == rM && rsub0 > a0) {
       q--;
-      a0 += b[0];
-      rM += b[1] + (a0 >> 128);
-      a0 &= 2**128-1;
+      a0 += b0;
+      rM += b1 + (a0 >> 128);
+      a0 &= TWO_TO_THE_ONE_HUNDRED_TWENTY_EIGHT_MINUS_ONE;
     }
 
-    q += qM;
     uint256 r = (((rM - rsub21) << 128) + _am - rsub0) >> shift;
 
-    return (q, r);
+    return (q + (_aM / _b), r);
```

Comment:
I think I made a mistake here when removing `qM` to free up some stack space.
However, it falsely passes the tests :confused: 
This should be:
```suggestion
    return (q + (_aM / _b << 128), r);
```